src/basis/tracker.js

```diff
@@ -117,18 +213,235 @@ function checkShow(){
         type: 'ui',
         path: stringifyPath(getPathByNode(visibleElement)),
         selector: selector.selectorStr,
-        event: 'show',
+        event: SHOW_EVENT,
         data: selector.data
       });
   }
 }
 
-//
-// main API
-//
+/**
+ * Search for the first key in `obj` (`obj` can be a nested object)
+ * and replaces a value of the finded key with `value`
+ * private except for tests
+ * @param {Object} obj any object
+ * @param {string} sample some stub like '*'
+ * @param {string} value a dynamic value
+ * @return
+ */
+function setDeep(obj, sample, value){
+  for (var key in obj)
+  {
+    if (hasOwnProperty.call(obj, key))
+    {
+      if (obj[key] === sample)
+      {
+        obj[key] = value;
+        return;
+      }
+      else if (typeof obj[key] === 'object')
+        return setDeep(obj[key], sample, value);
+    }
+  }
+}
+
+/**
+ * Clean up string
+ * @param {string} value
+ * @private
+ * @return {string}
+ */
+function escapeQuotes(value){
+  return String(value).replace(/\"/g, '\\"');
+}
+
+/**
+ * Search for the first key in `obj` (`obj` can be a nested object)
+ * and replaces a value of the finded key with `value`
+ * private except for tests
+ * @param {Array.<RoleObject>} path
+ * @param {Object} item custom tracking data object from some tracking map
+ * @param {Object} item.data
+ * @param {Array.<string>} item.selector
+ * @param {string} item.selectorStr
+ * @param {DOMEvent} event
+ * @return {{ async: boolean, datatoTrack: TrackingDataObject }}
+ */
+function handleEventFor(path, item, event){
+  if (!isPathMatchSelector(path, item.selector))
+    return;
+
+  var data = basis.object.slice(item.data);
+  var INPUT_EVENTS = ['keyup', 'keydown', 'keypress', 'input']; // when a user is typing
+
+  if (INPUT_EVENTS.indexOf(event.type) != -1)
+  {
+    data.inputValue = event.target.value;
+
+    return {
+      async: true,
+      dataToTrack: {
+        type: 'ui',
+        path: stringifyPath(path),
+        selector: stringifyPath(item.selector),
+        event: event.type,
+        data: typeof data.dataToTrackFn === 'function' ? data.dataToTrackFn(data, event) : data
```

Comment:
1. It would be better to normalize field on `item` create phase:
```js
item.data.dataToTrackFn = typeof ... === 'function' ? ... : basis.fn.$self;
```
In this case we don't need duplicate logic to check the field, just `data.dataToTrackFn(data, event)`.

2. I believe we need better name for `dataToTrackFn`